modules/mobile/app/controllers/mobile/v0/locations_controller.rb

```diff
@@ -7,10 +7,16 @@ module V0
     class LocationsController < ApplicationController
       def show
         Rails.logger.info('Mobile Location', param_id: params[:id])
+
         lh_location = service.get_location(params[:id])
         Rails.logger.info('Mobile Location, Lighthouse Info', lh_location: lh_location)
-        id = lh_location[:identifier].first[:value][4..]
+        if lh_location[:identifier].nil?
+          raise Common::Exceptions::BackendServiceException, 'validation_errors_bad_request'
+        end
+
+        id = lh_location[:identifier].first[:value][4..][0..2]
```

Comment:
Don't know if you could use one range here? e.g. `lh_location[:identifier].first[:value][4..6]` (assuming the id is always in the same spot).

Reply:
Yup, that makes more sense